core/src/main/java/org/jruby/java/invokers/RubyToJavaInvoker.java

```diff
@@ -451,14 +457,21 @@ protected final T findCallableArityZero(IRubyObject self, String name) {
     protected final T findCallableArityOne(IRubyObject self, String name, IRubyObject arg0) {
         T callable = this.javaCallable;
         if ( callable == null ) {
+            if ( javaCallables[1].length == 1 ) {
+              callable = javaCallables[1][0];
+              if (!callable.isVarArgs()) checkCallableArity(callable, 1);
+              return callable;
+            }
```

Comment:
think the logic should be moved, once we're sure `javaCallables[1]` is valid

probably having an `else` statement bellow the 
`if ( javaCallables.length <= 1 || (callablesForArity = javaCallables[1]) == null ) {` check